### PR TITLE
don't use regex matching

### PR DESCRIPTION
**Pull request by tadly**
## Description

If the path within `${PROJECT_SOURCE_DIR}`contains special characters (e.g. `+`), the build will fail as they get interpreted by regex
## Types of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by Tadly** [comment]
Sorry it took this long.
I accidentally PR'ed against my own repo and forgot about it _face palms_

@Rechi you'll probably remember this :)

File: cmake/scripts/common/AddonHelpers.cmake [hunk 1]
```diff
@@ -52,7 +52,7 @@ macro (build_addon target prefix libs)
     # Read used headers from addon, needed to identitfy used kodi addon interface headers
     if(${prefix}_HEADERS)
       # Add the used header files defined with CMakeLists.txt from addon itself
-      if(${prefix}_HEADERS MATCHES ${PROJECT_SOURCE_DIR})
+      if(${prefix}_HEADERS STREQUAL ${PROJECT_SOURCE_DIR})
```
Comment (Rechi) [inline cmake/scripts/common/AddonHelpers.cmake]: @AlwinEsch was your intention to use  `MATCHES` as exact string match or some kind of StartsWith, which cmake doesn't have?

Comment (AlwinEsch) [reply cmake/scripts/common/AddonHelpers.cmake]: No, was here during test the way who has worked :).
If "STREQUAL" can do the same it is also OK. Has not thought about circumstance with "+" in path.

**Comment by Tadly** [comment]
Updated as of rechis request :)
Sorry it took so long

**Comment by Rechi** [comment]
jenkins error unrelated, as this code isn't executed
build addons for windows and osx and everything still working

**Pull request merged**